runtime/interpreter/interpreter.go

```diff
@@ -5232,19 +5200,7 @@ func (interpreter *Interpreter) invalidateReferencedResources(value Value, destr
 	}
 
 	for value := range values { //nolint:maprange
-		switch value := value.(type) {
-		case *CompositeValue:
-			value.dictionary = nil
-			value.isDestroyed = destroyed
-		case *DictionaryValue:
-			value.dictionary = nil
-			value.isDestroyed = destroyed
-		case *ArrayValue:
-			value.array = nil
-			value.isDestroyed = destroyed
-		default:
-			panic(errors.NewUnreachableError())
-		}
+		value.Value = nil
```

Comment:
Here we just clear the referenced value instead of clearing the nested inner atree value of the referenced value.
So if a reference has a `nil` inne-value, then it's an invalidated reference.